core/src/main/resources/hudson/model/Messages.properties

```diff
@@ -244,7 +244,9 @@ Slave.Launching={0} Launching slave agent
 Slave.Network.Mounted.File.System.Warning=Are you sure you want to use network mounted file system for FS root? Note that this directory does not need to be visible to the master.
 Slave.Remote.Director.Mandatory=Remote directory is mandatory
 Slave.Terminated={0} slave agent was terminated
-Slave.the_remote_root_must_be_an_absolute_path=The remote root must be an absolute path.
+Slave.Remote.Relative.Path.Warning=Are you sure you want to use a relative path for the FS root? Note that relative \
+  paths require that you can assure that the selected launcher provides a consistent current working directory. Using \
```

Comment:
If you use JNLP to launch, the CWD depends on where you run the command, so (apart from our jnlp cloudy slaves) you probably should not use a relative path with jnlp (unless using jnlp cloudy slaves where it is very very useful)

With ssh the CWD will typically be the user's home directory and this is consistent so relative paths are very safe with ssh

Sent from my iPhone

> On 17 Mar 2015, at 13:56, Jesse Glick <notifications@github.com> wrote:
> 
> In core/src/main/resources/hudson/model/Messages.properties:
> 
> > @@ -244,7 +244,9 @@ Slave.Launching={0} Launching slave agent
> >  Slave.Network.Mounted.File.System.Warning=Are you sure you want to use network mounted file system for FS root? Note that this directory does not need to be visible to the master.
> >  Slave.Remote.Director.Mandatory=Remote directory is mandatory
> >  Slave.Terminated={0} slave agent was terminated
> > -Slave.the_remote_root_must_be_an_absolute_path=The remote root must be an absolute path.
> > +Slave.Remote.Relative.Path.Warning=Are you sure you want to use a relative path for the FS root? Note that relative \
> > +  paths require that you can assure that the selected launcher provides a consistent current working directory. Using \
> I think Stephen is trying to say something about cloudy slaves that share a common filesystem despite apparently different launch details, but I am not really sure either. Deserves a clearer explanation.
> 
> —
> Reply to this email directly or view it on GitHub.
> 
> -- 
> You received this message because you are subscribed to the Google Groups "engineering-code-reviews" group.
> To unsubscribe from this group and stop receiving emails from it, send an email to engineering-code-reviews+unsubscribe@cloudbees.com.
> To post to this group, send email to engineering-code-reviews@cloudbees.com.
> To view this discussion on the web visit https://groups.google.com/a/cloudbees.com/d/msgid/engineering-code-reviews/jenkinsci/jenkins/pull/1606/r26573715%40github.com.